script/run-cypress-tests-circle.sh

```diff
@@ -4,13 +4,13 @@ if [ "$(find src -name '*.cypress.spec.js' | wc -l)" -eq 0 ]; then
   echo "No Cypress tests found."
   exit 0
 else
-  # Use the junit reporter & save results in './test-results'.
-  reporterArgs="--reporter junit --reporter-options \"mochaFile=test-results/e2e-test-output-[hash].xml\""
-    
+  # Use mocha-junit-reporter and save results in './test-results'.
+  reporterArgs="--reporter cypress-multi-reporters --reporter-options \"configFile=config/cypress-reporters.json\""
+
   # Use the CircleCI CLI to get Cypress tests & split them on different machines by timing. Commas are inserted as the delimiter for file paths.
   specArgs="--spec $(echo '"$(circleci tests glob "src/**/tests/**/*.cypress.spec.js" | circleci tests split --split-by=timings | paste -sd "," -)"')"
 
   # Start the web server & run Cypress tests.
   # Wait for http://localhost:3001/generated/style.css to finish bundling so Cypress doesn't start too soon.
-  yarn start-server-and-test watch http-get://localhost:3001/generated/style.css "cypress run $reporterArgs $specArgs"
+  yarn start-server-and-test "node src/platform/testing/e2e/test-server.js --buildtype=vagovprod --port=3001 > /dev/null 2>&1" :3001 "yarn cy:run $reporterArgs $specArgs"
```

Comment:
In the HCA example, I noticed that the tests [begin](https://circle-production-customer-artifacts.s3.amazonaws.com/picard/5a947641c9e77c00018484ce/5ee24f71fd7aa57cd91916df-1-build/artifacts/cypress/videos/applications/hca/tests/hca.cypress.spec.js.mp4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20200611T190015Z&X-Amz-SignedHeaders=host&X-Amz-Expires=60&X-Amz-Credential=AKIAJR3Q6CR467H7Z55A%2F20200611%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=9158ecaf745d5fb7ab93e6118c60528ccc6c3d9320697faa0cb4a79d711e71a8) running way before the server is spun up. 

Originally I had the `start-server-and-test` command waiting for `generated/style.css` when we were using `watch`. I am wondering if there is a similar solution when spinning up the test server here.

Reply:
I think it mattered with `yarn watch` because it could still be bundling things. This just serves up files that have already been generated, so as soon as the base URL becomes available, presumably everything else will be accessible as well.

Reply:
I'm wondering if there is a way we can make it wait for the URL to become available before running the tests. That would shave off close to 30s in the beginning of each test.

Reply:
I was thinking that the delay in the beginning of the video was being added to the test time. But it turns out it isn't so this is okay!